src/hooks/useCurrentLocation.jsx

```diff
@@ -0,0 +1,35 @@
+function listErrors(error) {
+  switch (error.code) {
+    case error.PERMISSION_DENIED:
+      console.error('User denied the request for Geolocation.');
+      break;
+    case error.POSITION_UNAVAILABLE:
+      console.error('Location information is unavailable.');
+      break;
+    case error.TIMEOUT:
+      console.error('The request to get user location has timed out.');
+      break;
+    case error.UNKNOWN_ERROR:
+      console.error('An unknown error occurred.');
+      break;
+  }
+}
+
+export function useCurrentLocation() {
```

Comment:
this is not a hook if you don't use hooks inside

useCurrentLocation-> getCurrentLocation